### PR TITLE
Introduce template operator pipeline

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**

Changes proposed in this pull request:

- Introduce pipelines for template operator
- Rename pipelines of lifecycle- and module-manager
    - Since prow does not distinguish between Repositories in the job history if the y have the same naming
    - Example: https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/pr-logs/directory/pull-lint
    This history includes jobs for lifecycle-manager, module-manager, as well as runtime-watcher, because the jobs have the same name

**Related issue(s)**

### CODE REVIEW

File: prow/jobs/module-manager/module-manager.yaml [hunk 1]
```diff
@@ -3,13 +3,13 @@
 
 presubmits: # runs on PRs
   kyma-project/module-manager:
-    - name: pull-lint
+    - name: pull-module-mgr-lint
```
Comment (adityabhatia) [inline prow/jobs/module-manager/module-manager.yaml:6]: is this rename required, since pull-lint job was in `module-manager.yaml`

**Comment by adityabhatia** [comment]
looks good overall, but do we require the new job names? IMO the previous job names were more understandable.

File: prow/jobs/module-manager/module-manager.yaml (hunk 1 above)
Comment (jeremyharisch) [reply prow/jobs/module-manager/module-manager.yaml:6]: Please have a look at my description
```
- Since prow does not distinguish between Repositories in the job history if they have the same naming
- Example: https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/pr-logs/directory/pull-lint
This history includes jobs for lifecycle-manager, module-manager, as well as runtime-watcher, because the jobs have the same name
```

I personally prefer if we could have separated PR histories for the repos, to have a better overview in the job history.

Comment (adityabhatia) [reply prow/jobs/module-manager/module-manager.yaml:6]: Thanks for the clarification

**Review by adityabhatia — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Comment by jakobmoellersap** [comment]
@jeremyharisch Why was this merged without adjusting the github actions in lifecycle manager and module manager? I explicitly told you that the jobs need to be adjusted with a dedicated pr. Now all build job waits will fail until this is adjusted.

**Comment by jeremyharisch** [comment]
@jakobmoellersap here are the dedicated PRs:
https://github.com/kyma-project/lifecycle-manager/pull/309
https://github.com/kyma-project/module-manager/pull/169
